agent/profile

```diff
@@ -45,3 +58,13 @@ _pbench_setup
 
 unset -f _pbench_pathins
 unset -f _pbench_setup
+
+# If this script was invoked with any arguments, execute them as a command line,
+# INSTEAD of returning to our caller; otherwise, we just return.
+#
+# This allows us to use this script as the entrypoint to the Agent containers,
+# such that `podman run <cid> pbench-results-push` will execute this script and
+# then run the provided command.
+if [[ -n "$1" ]] ; then
+    exec "${@}"
```

Comment:
Profiles are "sourced" when installed by the operating system in `/etc/profile.d`, so it would be better to avoid duplication and just have an "entry point" script which sourced the profile and executed the first argument.

Reply:
That's certainly a valid approach, but it means creating yet another little script which we have to manage.  I liked this approach because it doesn't change the behavior in the existing use of `agent/profile`, and it doesn't require any new files.

>  it would be better to avoid duplication

What is being duplicated?

Reply:
First off, how will line 69 work in normal operation?  When you `source` a script, the command line arguments of your "sourcer" are available to the script being sourced, no?

Second, we are not changing everything to be container based.  We need to keep these use cases separate.

Let's not merge them.

Reply:
And, this is contrib work, which should not be changing the base mode of the RPM operation just to make a contrib behavior work.

I think the principle of separation of concerns applies here.

One concern is how we package our software in an RPM for installation.  That RPM has to support whatever environment it is installed in, so we cannot assume a containerized environment for the RPM.

The other concern is a containerized environment.  That environment can easily layer on top of the RPM to address its concerns.

Merging those two concerns together makes little sense until we are removing one concern, which we are not.